dvc/config.py

```diff
@@ -277,8 +277,8 @@ def load(self, validate=True):
         self.clear()
         self.update(conf)
 
-        # Add resolved default cache.dir
-        if not self["cache"].get("dir"):
+        # Add resolved defa1ult cache.dir
+        if not self["cache"].get("dir") and self.dvc_dir:
```

Comment:
I'm really not sure if we should load the configs even if it's not a DVC directory.

Reply:
@skshetry Could you elaborate please? You mean that we should call it on-premise when some field in config is being accessed?

Reply:
I mean, should we try to load config if it's not a DVC repository? Currently, this function that we are discussing loads from all the local, system and global configs even though `self.dvc_dir` is not set (i.e. it's not a DVC dir). What I am asking is, are those configs used anywhere when it's not a DVC dir?

Reply:
Is there any point to try to load those?

Reply:
@skshetry Ah, got it. Yes, those configs (global and system) are indeed used. For example in analytics.py. Imagine running `dvc  get` outside of dvc repo, where dvc will try to send analytics event and so needs to check if analytics are disabled in the config or not.

Reply:
Got it, thanks.